quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs

```diff
@@ -258,8 +258,8 @@ fn deserialize_mapping_type(
             #[allow(clippy::collapsible_if)]
             if !text_options.indexed {
                 if text_options.tokenizer.is_some()
-                    || text_options.record == IndexRecordOption::Basic
-                    || !text_options.fieldnorms
+                    || text_options.record != IndexRecordOption::Basic
```

Comment:
With this check, `indexed: false, record: basic` will not fail. It should. `text_options.record` should probably be an option.

Reply:
By default, indexed is set to true. If the IndexRecordOption is an option, what should the default value be since it can't be None ?

Reply:
To be more accurate, at first I had already changed to IndexRecordOption to None, this caused issues since there was the possibility of having a text that could be indexed but there were no RecordOptions.

Reply:
> `None`. If `indexed` is `false` and `record` is `Some(...)`, then that's an error. If `indexed` is true and `record` is `None`, then we use the default value `IndexRecordOption::Basic`.

I will update the code then.

Reply:
I have no preference on this so let's follow @guilload advice :)

As a side note, if we put an Option on record, why not put an option on fieldnorm and tokenizer too? (we don't need to do that now).

Reply:
Yes, `tokenizer` and `record` should be options. `fieldnorms` is a boolean, so not really :)

Reply:
`record` is now an option, `tokenizer` was already one so no changes needed there.